beacon_node/network/src/network_beacon_processor/gossip_methods.rs

```diff
@@ -1328,13 +1328,19 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
                 return None;
             }
             Err(e @ BlockError::BlobNotRequired(_)) => {
-                // TODO(das): penalty not implemented yet as other clients may still send us blobs
-                // during early stage of implementation.
                 debug!(self.log, "Received blobs for slot after PeerDAS epoch from peer";
                     "error" => %e,
                     "peer_id" => %peer_id,
                 );
                 self.propagate_validation_result(message_id, peer_id, MessageAcceptance::Ignore);
+                // Receiving blobs when not required just spends unnecessary bandwidth of us. If the
+                // peer is malicious the damange is mild. If the peer is faulty, it's likely due to
+                // a non-updated node which will be quickly disconnected due to other faults.
+                self.gossip_penalize_peer(
+                    peer_id,
+                    PeerAction::MidToleranceError,
+                    "blob_not_required",
+                );
```

Comment:
Implemented the penalty, all implementations should not send blobs by now @jimmygchen

Reply:
We already penalise peers for `BlobNotRequired` here:
https://github.com/sigp/lighthouse/blob/22c68ddf7de43e1d449c047466328493957de3b8/beacon_node/network/src/network_beacon_processor/gossip_methods.rs#L973-L987

Here's Pryms getting downscored in devnet 

```
Jan 23 10:15:00.128 DEBG Invalid gossip blob                     blob_index: 2, block_slot: 69, block_root: 0xcb9e88ec1f0505bbe527a7e2664bcf34e85c62943a636d74aabfe159c4aa2b52, outcome: BlobNotRequired(Slot(69))
Jan 23 10:15:00.128 DEBG Peer score adjusted                     score: -19.84, peer_id: 16Uiu2HAm3Dq3c3ubEQihCSFQ2dGu6TWwzazCTRwsS33AjtbENz8a, msg: bad_gossip_blob_ssz, service: libp2p
```

Why do we handle this variant in process gossip block here? Is it even possible to hit this?

Reply:
Oh right, it's unreachable here. Fixed